resource-validator/src/main/scala/com/broadinstitute/dsp/resourceValidator/DbReader.scala

```diff
@@ -36,27 +36,31 @@ object DbReader {
     sql"""select googleProject, initBucket from CLUSTER WHERE status="Deleted";"""
       .query[InitBucketToRemove]
 
+  val deletedRuntimeQuery =
+    sql"""SELECT distinct c1.id, googleProject, clusterName, rt.cloudService, c1.status from CLUSTER AS c1
+             INNER join RUNTIME_CONFIG AS rt ON c1.`runtimeConfigId`=rt.id WHERE c1.status="Deleted"
+             and NOT EXISTS (SELECT * from CLUSTER as c2 where c2.googleProject = c1.googleProject
+             and c2.clusterName=c1.clusterName and (c2.status="Stopped" or c2.status="Running"));"""
+      .query[Runtime]
+
+  val erroredRuntimeQuery =
+    sql"""SELECT DISTINCT c1.id, googleProject, clusterName, rt.cloudService, c1.status from CLUSTER AS c1
+             INNER join RUNTIME_CONFIG AS rt ON c1.`runtimeConfigId`=rt.id WHERE c1.status="Error"
+             and NOT EXISTS (SELECT * from CLUSTER as c2 where c2.googleProject = c1.googleProject
+             and c2.clusterName=c1.clusterName and (c2.status="Stopped" or c2.status="Running"));"""
+      .query[Runtime]
```

Comment:
I wonder if it'd be safer to use `LEFT/RIGHT JOIN` instead of `INNER JOIN` so we could catch runtimes with no corresponding runtimeConfig in case such an aberration were to occur.

Reply:
if somehow that occurs, we won't be able to know what resource we created either cuz `RUNTIME_CONFIG` is where we persist whether this is a gce or dataproc runtime

Reply:
ah yea, you're right

Reply:
I think there's foreign key on runtimeconfig, which should enforce that